lib/src/Base/IterativeStat/IterativeMoments.cxx

```diff
@@ -0,0 +1,301 @@
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#include "openturns/IterativeAlgorithm.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/IterativeMoments.hxx"
+#include "openturns/SpecFunc.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(IterativeMoments)
+
+static const Factory<IterativeMoments> Factory_IterativeMoments;
+
+/* Default constructor */
+IterativeMoments::IterativeMoments(const UnsignedInteger orderMax, const UnsignedInteger dimension)
+  : IterativeAlgorithmImplementation(dimension)
+  , orderMax_(orderMax)
+  , centeredMoments_(orderMax, dimension_)
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+IterativeMoments * IterativeMoments::clone() const
+{
+  return new IterativeMoments(*this);
+}
+
+/* String converter */
+String IterativeMoments::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << IterativeMoments::GetClassName()
+      << " iteration=" << iteration_
+      << " dimension=" << dimension_
+      << " order max=" << orderMax_
+      << " centered moments=" << centeredMoments_.__repr__();
+  return oss;
+}
+
+/* String converter */
+String IterativeMoments::__str__(const String & offset) const
+{
+  return getCenteredMoments().__str__(offset);
+}
+
+/*  Accessor to the centered moments */
+Sample IterativeMoments::getCenteredMoments() const
+{
+  return centeredMoments_;
+}
+
+/* Accessor to the maximum order declared */
+UnsignedInteger IterativeMoments::getOrder() const
+{
+  return orderMax_;
+}
+
+Point IterativeMoments::getMomentOne()
+{
+  if (iteration_ >= 1)
+  {
+    return centeredMoments_[0] / iteration_;
+  }
+  else
+  {
+    return centeredMoments_[0];
+  }
+}
+
+/* Accessor to the mean */
+Point IterativeMoments::getMean() const
+{
+  if (!(iteration_ > 0)) throw InternalException(HERE) << "Error: cannot compute the mean of an empty sample.";
+
+  return centeredMoments_[0] / iteration_;
+}
+
+/* Accessor to the variance */
+Point IterativeMoments::getVariance() const
+{
+  if (!(iteration_ > 0)) throw InternalException(HERE) << "Error: cannot compute the variance per component of an empty sample.";
+  if (!(orderMax_ >= 2)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum should be at least 2.";
+
+  /* Special case for a size 1 */
+  if (iteration_ == 1) return Point(dimension_, 0.0);
+
+  /* unbiased estimator */
+  return centeredMoments_[1] / (iteration_ - 1);
+}
+
+/* Accessor to the skewness */
+Point IterativeMoments::getSkewness() const
+{
+  if (!(iteration_ >= 2)) throw InternalException(HERE) << "Error: cannot compute the skewness per component of a sample of size less than 2.";
+  if (!(orderMax_ >= 3)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order should be at least 3.";
+
+  /* Special case for a size 2 */
+  if (iteration_ == 2) return Point(dimension_, 0.0);
```

Comment:
I am not sure about this one. This is because the skewness is a ratio. If the two observations are equal, then the variance is zero and the skewness is undefined. Otherwise, the variance can be computed and the skewness is not necessarily zero (but we cannot apply the bias correction).